src/main/java/ch/njol/skript/util/Timespan.java

```diff
@@ -32,97 +37,134 @@
 import ch.njol.util.coll.CollectionUtils;
 import ch.njol.yggdrasil.YggdrasilSerializable;
 
-/**
- * @author Peter Güttinger
- * @edited by Mirreducki. Increased maximum timespan.
- */
 public class Timespan implements YggdrasilSerializable, Comparable<Timespan> { // REMIND unit
 
-	private final static Noun m_tick = new Noun("time.tick");
-	private final static Noun m_second = new Noun("time.second");
-	private final static Noun m_minute = new Noun("time.minute");
-	private final static Noun m_hour = new Noun("time.hour");
-	private final static Noun m_day = new Noun("time.day");
-	private final static Noun m_week = new Noun("time.week");
-	private final static Noun m_month = new Noun("time.month");
-	private final static Noun m_year = new Noun("time.year");
-	final static Noun[] names = {m_tick, m_second, m_minute, m_hour, m_day, m_week, m_month, m_year};
-	final static long[] times = {50L, 1000L, 1000L * 60L, 1000L * 60L * 60L, 1000L * 60L * 60L * 24L,  1000L * 60L * 60L * 24L * 7L,  1000L * 60L * 60L * 24L * 30L,  1000L * 60L * 60L * 24L * 365L};
-	final static HashMap<String, Long> parseValues = new HashMap<>();
+
+	public enum TimePeriod {
+
+		TICK("time.tick", 50L),
+		SECOND("time.second", 1000L),
+		MINUTE("time.minute", SECOND.time * 60L),
+		HOUR("time.hour", MINUTE.time * 60L),
+		DAY("time.day", HOUR.time * 24L),
+		WEEK("time.week", DAY.time * 7L),
+		MONTH("time.month", DAY.time * 30L), // Who cares about 28, 29 or 31 days?
+		YEAR("time.year", DAY.time * 365L);
```

Comment:
couldn't we automate this first parameter?
`time + enumerator.name().toLowerCase(Locale.ENGLISH)`